### ISSUE TITLE
False positives when first sentence of Javadoc is on a separate line

### ISSUE THREAD

**Issue by patchwork01** [checkstyle/checkstyle#14689]
I have read check documentation: https://checkstyle.org/checks/javadoc/summaryjavadoc.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac TestCheckstyleJavadocSummary.java

/var/tmp $ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="SummaryJavadoc">
            <property name="forbiddenSummaryFragments" value="fail-summary-fragment"/>
        </module>
    </module>
</module>

/var/tmp $ cat TestCheckstyleJavadocSummary.java
public class TestCheckstyleJavadocSummary {

    /**
     * Summary sentence on its own line.
     * <p>
     * Another sentence that is not part of the summary, so this should not matter: fail-summary-fragment.
     */
    void foo1() {
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"

/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.14.2-all.jar -c config.xml TestCheckstyleJavadocSummary.java
Starting audit...
[ERROR] /var/tmp/TestCheckstyleJavadocSummary.java:3: Forbidden summary fragment. [SummaryJavadoc]
Audit done.
Checkstyle ends with 1 errors.
```

---

When a summary sentence is on its own line, it should still be processed as the entire summary. Further lines that contain forbidden summary fragments should not cause the check to fail.

In addition, looking at the code, it appears that if the summary sentence includes the default period character in the middle, it cuts off there. Any forbidden fragment in the rest of the sentence is ignored, and if the rest of the sentence does not contain the period character, that is ignored too. This is done based on the default period character irrelevant of whether a custom period character is configured.

---

I have a fix for this on my fork, and will raise a PR.

Here's an issue on another repository where this has caused false positives:
- https://github.com/gchq/sleeper/issues/2064

**Comment by romani** [comment]
@patchwork01 , thanks a lot for your contribution.
We can release it sooner if you need released fix sooner, in other case
We will release it at the end of month as usually. Let me know if need fix to be released sooner.
